### PR TITLE
Revert path quoting

### PR DESCRIPTION
**Pull request by uasi**
# Description

Fixes https://github.com/nushell/nushell/pull/5384#issuecomment-1114160697

# Tests

Make sure you've run and fixed any issues with these commands:

- [x] `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- [x] `cargo clippy --all --all-features -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- [x] `cargo build; cargo test --all --all-features` to check that all the tests pass

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -88,10 +88,12 @@ fn main() -> Result<()> {
...
                 "--commands" | "-c" => {
                     // FIXME: Use proper quoting. `escape_quote_string()` can't be used for now due to https://github.com/nushell/nushell/issues/5383.
-
                     args.next().map(|a| format!("`{}`", a))
                 }
-                "--config" | "--env-config" => args.next().map(|a| escape_quote_string(&a)),
+                "--config" | "--env-config" => {
+                    // FIXME: Use proper quoting. `escape_quote_string()` can't be used because, as Windows path, foo\bar.nu and "foo\\bar.nu" are not equal.
+                    args.next().map(|a| format!("`{}`", a))
+                }
```
Comment (uasi) [inline src/main.rs:96]: `foo\bar.nu` and `"foo\\bar.nu"` _should_ be equal, actually.

Comment (jntrnr) [reply src/main.rs:96]: it would be nice to find out why they're not equal. That seems like it could be the real fix.

Comment (jntrnr) [reply src/main.rs:96]: For the simple case, the seem to be the same:

```
> let-env FOO = "foo\\bar.nu"
> $env.FOO
foo\bar.nu
> let-env FOO = `foo\bar.nu`
> $env.FOO
foo\bar.nu

Comment (uasi) [reply src/main.rs:96]: I suspect the path argument is mistakenly parsed with `nu_parser::trim_quotes()` where `nu_parser::unescape_unquote_string()` should be used.

Comment (uasi) [reply src/main.rs:96]: Probably this:

https://github.com/nushell/nushell/blob/80d57d70cd48fa0fa40181eaa77fcddea4b3af75/crates/nu-parser/src/parser.rs#L1953-L1958

There seem to be a dozen of mistakes of this sort, though.

Comment (jntrnr) [reply src/main.rs:96]: I'd be happy to fix them

**Comment by jntrnr** [comment]
I fixed https://github.com/nushell/nushell/pull/5384#issuecomment-1114160697 and a bunch of related issues. As you pointed out, we were using quotes (or trimming quotes) in lots of places we shouldn't have been. I switched all the ones I could find that weren't command names to use your function, then went through all the places that take strings from outside nu and make sure those also worked okay.

Closing this as I think the issue is now fixed. Thanks for pushing on it.

**Pull request closed without merging**